File: src/main/scala/com/ajjpj/adiagram/render/shapes/lineend/NullLineEnd.scala
package com.ajjpj.adiagram.render.shapes.lineend

import javafx.scene.canvas.GraphicsContext
import com.ajjpj.adiagram.geometry.{Angle, APoint}
import com.ajjpj.adiagram.render.base.LineStyle
import com.ajjpj.adiagram.geometry.transform.Translation
import com.ajjpj.adiagram.ui.Zoom

/**
 * @author arno
 */
class NullLineEnd extends ALineEnd {
  override def shortenLengthUnzoomed(style: LineStyle) = 0.0
  override def width(style: LineStyle, zoom: Zoom) = 0.0

  override def paint(gc: GraphicsContext, p: APoint, angle: Angle, style: LineStyle, t: Translation, zoom: Zoom) {
  }
}

File: src/main/kotlin/com/ajjpj/adiagram/render/shapes/lineend/NullLineEnd.kt
package com.ajjpj.adiagram.render.shapes.lineend

import com.ajjpj.adiagram.Zoom
import com.ajjpj.adiagram.geometry.APoint
import com.ajjpj.adiagram.geometry.AffineTransformation
import com.ajjpj.adiagram.geometry.Angle
import com.ajjpj.adiagram.render.style.LineStyle
import javafx.scene.canvas.GraphicsContext

class NullLineEnd: ALineEnd {
    override fun shortenLengthUnzoomed(style: LineStyle): Double = .0
    override fun width(style: LineStyle, zoom: Zoom): Double = .0

    override fun paint(gc: GraphicsContext, p: APoint, angle: Angle, style: LineStyle, t: AffineTransformation, zoom: Zoom) {
    }
}